### ISSUE TITLE
Error using Python client with local storage, search filter and missing payload

### ISSUE THREAD

**Issue by todowa** [qdrant/qdrant-client#187]
I'm getting an error when running the following the code from [Quickstart](https://qdrant.tech/documentation/quick-start/) using the Python client and local storage. I'm using qdrant-client 1.2.0; python3.10.7.

```
from qdrant_client.http.models import PointStruct

operation_info = client.upsert(
    collection_name=COLLECTION_NAME,
    wait=True,
    points=[
        PointStruct(id=1, vector=[0.05, 0.61, 0.76, 0.74], payload={"city": "Berlin"}),
        PointStruct(id=2, vector=[0.19, 0.81, 0.75, 0.11], payload={"city": ["Berlin", "London"]}),
        PointStruct(id=3, vector=[0.36, 0.55, 0.47, 0.94], payload={"city": ["Berlin", "Moscow"]}),
        PointStruct(id=4, vector=[0.18, 0.01, 0.85, 0.80], payload={"city": ["London", "Moscow"]}),
        PointStruct(id=5, vector=[0.24, 0.18, 0.22, 0.44], payload={"count": [0]}),
        PointStruct(id=6, vector=[0.35, 0.08, 0.11, 0.44]),
    ]
)
```

Then later:
```
from qdrant_client.http.models import Filter, FieldCondition, MatchValue
search_result = client.search(
    collection_name="test_collection",
    query_vector=[0.2, 0.1, 0.9, 0.7],
    query_filter=Filter(
        must=[
            FieldCondition(
                key="city",
                match=MatchValue(value="London")
            )
        ]
    ),
    limit=3
)
```

The error is:

```
File ~/projects/.../.venv/lib/python3.10/site-packages/qdrant_client/qdrant_client.py:283, in QdrantClient.search(self, collection_name, query_vector, query_filter, search_params, limit, offset, with_payload, with_vectors, score_threshold, append_payload, consistency, **kwargs)
    199 def search(
    200     self,
    201     collection_name: str,
   (...)
    217     **kwargs: Any,
...
---> 29     if k not in data:
     30         return
     31     value = data[k]

TypeError: argument of type 'NoneType' is not iterable
```

The error seems to be because neither `qdrant_client.local.payload_filters.calculate_payload_mask` nor `qdrant_client.local.payload_value_extractors.value_by_key` can handle missing (`None`) payloads, yet this is the case for `PointStruct` with `id=6`.

Not sure what is most elegant but changing `if check_filter(payload_filter, payload, ids_inv[i]):` to `if payload is not None and check_filter(payload_filter, payload, ids_inv[i]):` in line 174 of `qdrant_client.local.payload_filters.calculate_payload_mask` appears to resolve the issue.

**Comment by joein** [comment]
Hi @todowa
Thanks for pointing that out, we will fix it soon

**Comment by joein** [comment]
#191 fixes it, however that PR is not complete, due to #192

**Comment by joein** [comment]
Should be fixed as of `qdrant-client==1.3.0`
